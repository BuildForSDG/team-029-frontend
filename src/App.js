import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

// pages import
import LandingPage from './pages/landing/Landing.page.jsx';
import LoginPage from './pages/login/Login.jsx';
import Dashboard from './pages/Dasboard/Dashboard.jsx';

import withNavbar from './components/withNavbar/withNavbar.js'

import './App.scss';
const App = () => {
	return (
		<main id="app">
			<Router>
				<Switch>
					<Route path="/" exact render={(props) => withNavbar(LandingPage, { componentProps: props })} />
					<Route path="/login" render={props => withNavbar(LoginPage, { navProps: { hasLinks: false }, componentProps: props })} />
					<Route path="/dashboard" component={Dashboard} />
				</Switch>
			</Router>
		</main>
	);
};

export default App;
