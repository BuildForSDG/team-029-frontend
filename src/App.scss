
$background: #c4c4c4;

.app{
    background: $background;
    min-width: 100vw;
    min-height: 100vh;
    padding: 4rem 2rem;
}