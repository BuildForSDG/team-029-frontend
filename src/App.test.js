import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('renders welcome to our app', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/Hello welcome to our app/i);
  expect(linkElement).toBeInTheDocument();
});
