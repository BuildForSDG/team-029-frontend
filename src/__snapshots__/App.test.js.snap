// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders welcome to our app 1`] = `
<DocumentFragment>
  <div>
    <div
      class="nav-container"
    >
      <div
        class="logo-container"
      >
        <img
          alt="logo"
          src="logo.png"
        />
      </div>
      <ul>
        <li
          class="active"
        >
          <a>
            Home
          </a>
        </li>
        <li>
          <a>
            Road Info
          </a>
        </li>
      </ul>
    </div>
    <div
      class="main"
    >
      <div
        class="left-screen"
      >
        <div
          class="caption"
        >
          <h3>
            Emergency Alert 
            <span>
              !
            </span>
          </h3>
          <p>
            With a single click, A road worden would attend to your emergency
          </p>
          <p
            class="tips"
          >
            <span
              class="bar"
            >
              |
            </span>
             Accident 
            <span
              class="bar"
            >
              |
            </span>
             Robery 
            <span
              class="bar"
            >
              |
            </span>
             Car Breakdown
          </p>
          <div
            class="btns"
          >
            <button
              class="ui button custom-btn"
            >
              Send an Alert
            </button>
            <h6>
              Or
            </h6>
            <button
              class="ui button custom-btn outline"
            >
              View Road Info
            </button>
          </div>
        </div>
        <div
          class="blob"
        >
          <img
            alt="blob"
            src="blob.png"
          />
        </div>
      </div>
      <div
        class="right-screen"
      >
        <div
          class="carImg"
        >
          <img
            alt="car"
            src="car.png"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
